activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -981,6 +995,12 @@ def build_arel(aliases)
         arel.from(build_from) unless from_clause.empty?
         arel.lock(lock_value) if lock_value
 
+        with_annotation_values.reject(&:nil?).tap do |annotations|
```

Comment:
`tap` here is confusing an it would be better to just assign to a variable since the value itself is not returned.